db/interfaces.go

```diff
@@ -49,48 +49,47 @@ type DatabaseMap interface {
 	OneSelector
 	Inserter
 	SelectExecer
-	Begin() (Transaction, error)
+	BeginTx(context.Context) (Transaction, error)
 }
 
 // Executor offers the full combination of OneSelector, Inserter, SelectExecer
-// and adds a handful of other high level Gorp methods we use in Boulder.
+// and adds a handful of other high level borp methods we use in Boulder.
 type Executor interface {
 	OneSelector
 	Inserter
 	SelectExecer
-	Delete(...interface{}) (int64, error)
-	Get(interface{}, ...interface{}) (interface{}, error)
-	Update(...interface{}) (int64, error)
-	Query(string, ...interface{}) (*sql.Rows, error)
+	Queryer
+	Delete(context.Context, ...interface{}) (int64, error)
+	Get(context.Context, interface{}, ...interface{}) (interface{}, error)
+	Update(context.Context, ...interface{}) (int64, error)
 }
 
-// Queryer offers the Query method. Note that this is not read-only (i.e. not
-// Selector), since a Query can be `INSERT`, `UPDATE`, etc. The difference
-// between Query and Exec is that Query can return rows. So for instance it is
+// Queryer offers the QueryContext method. Note that this is not read-only (i.e. not
+// Selector), since a QueryContext can be `INSERT`, `UPDATE`, etc. The difference
+// between QueryContext and ExecContext is that QueryContext can return rows. So for instance it is
 // suitable for inserting rows and getting back ids.
 type Queryer interface {
-	Query(string, ...interface{}) (*sql.Rows, error)
+	QueryContext(context.Context, string, ...interface{}) (*sql.Rows, error)
 }
 
-// Transaction extends an Executor and adds Rollback, Commit, and WithContext.
+// Transaction extends an Executor and adds Rollback and Commit
 type Transaction interface {
 	Executor
 	Rollback() error
 	Commit() error
-	WithContext(ctx context.Context) gorp.SqlExecutor
 }
 
 // MappedExecutor is anything that can map types to tables, and which can
 // produce a SqlExecutor bound to a context.
```

Comment:
This comment is now out-of-date, it cannot "produce a SqlExecutor bound to a context" anymore.